core/internal/example/module.go

```diff
@@ -0,0 +1,126 @@
+package example
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/core/extension"
+	"github.com/gogo/protobuf/proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cosmos/cosmos-sdk/container"
+	"github.com/cosmos/cosmos-sdk/core/app"
+	"github.com/cosmos/cosmos-sdk/core/blockinfo"
+	"github.com/cosmos/cosmos-sdk/core/event"
+	"github.com/cosmos/cosmos-sdk/core/module"
+	"github.com/cosmos/cosmos-sdk/core/store"
+)
+
+// go:embed proto_image.bin.gz
+var pinnedProtoImage embed.FS
```

Comment:
The pinned image is what version of proto files the module itself was built against. This could be very different from the api module version available at init genesis. So far I see no easy way around the pinned image that ensures correct unknown field rejection